src/web/releases.rs

```diff
@@ -497,6 +497,69 @@ impl Default for Search {
     }
 }
 
+fn redirect_to_random_crate(req: &Request, conn: &mut PoolClient) -> IronResult<Response> {
+    // since there is a chance of this query returning an empty result,
+    // we retry a certain amount of times, and log a warning.
+    for _ in 0..20 {
```

Comment:
It seems strange to use a fallible query, is there not a way to make it always work?

Reply:
@Kixiron I tried (I think) 5-10 different approaches, the ones that always work are either too slow or involve things like a recursive CTE (which is in in the end similar to retrying here, while this is more readable IMHO).

Reply:
I just ran some numbers: current query runs between 2 and 8 ms (~) 
and delivers a random crate directly in most cases (50 misses for 10k queries), 

the repetitions are more or less a safeguard. 

Also I could not test on a production dataset so I don't know the amount of gaps in the id-span on the crate-table (and the production github stars without backfilling everything locally)